server/server.go

```diff
@@ -1866,6 +1872,12 @@ func (s *Server) ID() string {
 }
 
 func (s *Server) startGoRoutine(f func()) {
+	if checkStartGoRoutineFunc {
+		name := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
+		if strings.HasSuffix(name, "-fm") {
```

Comment:
This stable across releases?

Reply:
You mean go releases, right? not sure. the alternative is I could check for end of name being `.funcxxx`. It is sometimes `.func2()` or similar. I could dig a bit more.
Works for go 1.11 and 1.12. We could adapt in the future too.

Reply:
ok, just not sure how important it is to have the check etc.

Reply:
I can definitively remove check (and therefore test). Let me know if you want me to do that.

Reply:
For now I would.

Reply:
Ok removed check and test. Did a push --force.